controllers/front/ProductController.php

```diff
@@ -985,6 +992,45 @@ function ($elem) use ($checkedIdProductAttribute) {
             );
 
             if (empty($availableProductAttribute) && count($availableProductAttributes)) {
+                // if selected combination is NOT available ($availableProductAttribute) but they are other alternatives ($availableProductAttributes), then we'll try to get the closest.
```

Comment:
I think the best way to understand is to split this code into small methods. :thinking: